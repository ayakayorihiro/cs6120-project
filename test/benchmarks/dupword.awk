# ARGS: dupword_test.txt
# dupword -- find duplicate words in text
# Arnold Robbins, arnold@gnu.ai.mit.edu, Public Domain
# December 1991

{
    $0 = tolower($0)
    gsub(/[^A-Za-z0-9 \t]/, "");
    if ($1 == prev)
        print(FILENAME ":" FNR ": duplicate " $1 "\n")
    for (i = 2; i <= NF; i++)
        if ($i == $(i-1))
            printf(FILENAME ":" FNR ": duplicate " $i "\n")
    prev = $NF
}
