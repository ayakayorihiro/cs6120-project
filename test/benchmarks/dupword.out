1: duplicate hello
3: duplicate this
4: duplicate maybe
4: duplicate this
6: duplicate mistakes
